### ISSUE TITLE
PSL updates

### ISSUE THREAD

**Issue by alternativc** [authelia/authelia#8074]
### Description

PSL (https://publicsuffix.org/) updates are extremely behind in authelia. Domains removed from the list may take months to become eligeble for protection under authelia. Can the list be updated on each authelia build?

### Use Case

A domain is listed on PSL. The domain is removed, authelia should acknowledge that.

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] I have checked for related issues and checked the documentation

**Comment by Crowley723** [comment]
Authelia uses the golang.org/x/net/publicsuffix library to validate user defined domains.

The publicsuffix subpackage is updated as part of the x/net releases. We automatically update packages when updates become available. Looking at x/net/publicsuffix, new releases usually come out the first week of every month.

It's unlikely that we would fork the library to maintain our own version of the list.

Do you have any instances that show that Authelia is significantly behind the updated public suffix library?

**Comment by alternativc** [comment]
Sure. Here is a merged removal from the list from July that is still present in the latest authelia build: https://github.com/publicsuffix/list/pull/2055

**Comment by james-d-elliott** [comment]
That would indicate it's absent upstream as well. Is there someone trying to use one of these domains not on the current go list?

**Comment by Crowley723** [comment]
We pull the latest x/net/publicsuffix releases very rapidly after they are released.

As this issue is due to the upstream libraries' version of the publicsuffix list being out of date, I might recommend opening an issue with them.

The last time that their version was updated was [09/08/2023](https://cs.opensource.google/go/x/net/+/master:publicsuffix/data/;bpv=1)

**Comment by alternativc** [comment]
Thanks for pointing that out. Let me open an issue with them and close this one. Is there an authelia release train to watch out for?

**Comment by james-d-elliott** [comment]
We are about to start the 4.39.0 release cycle but we're be willing to cut a release for this provided we have not started merging features.

**Comment by nightah** [comment]
@alternativc I'm going to assume you're using the container deployment method, if so could you please try the `fix-psl` tag on DockerHub or GHCR?

**Comment by alternativc** [comment]
Running `authelia/authelia:fix-psl`. The issue still persists though:
```
time="2024-10-28T10:28:14Z" level=error msg="Configuration: session: domain config #1 (domain 'shiftcrypto.dev'): option 'domain' is not a valid cookie domain: the domain is part of the special public suffix list" stack="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.1/command.go:970                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\ninternal/runtime/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1700                                         goexit"
```

Can be reproduced by setting session cookies domain to `shiftcrypto.dev`

**Comment by james-d-elliott** [comment]
Try repulling it, should be good now.

**Comment by nightah** [comment]
Given that the branch would have now been auto-removed you can run `master` for now.
We will aim to cut a new patch release in the next few days including the fix.

**Comment by alternativc** [comment]
Thanks. Was just looking at that. Deployed master and works like a charm!

**Comment by nightah** [comment]
> Thanks. Was just looking at that. Deployed master and works like a charm!

This fix has been released under `v4.38.17` so you don't need to continue using the `master` tag.
